src/sampler.js

```diff
@@ -0,0 +1,8 @@
+export default class DefaultSampler {
+    isSample(span, parent) {
```

Comment:
per the previous discussion (which is lost due to the squash): IMO it's not necessary to consider whether a non-root span (i.e., a span with a parent) should be sampled in BasicTracer... every tracing system I'm aware of just inherits the parent's sampling status in that case.

Also, `s/isSample/isSampled/`... and please add a brief comment explaining the semantics and expected parameter and return types.